tests/test_perf_tracking.py

```diff
@@ -1982,15 +1992,112 @@ def test_empty_positions(self):
             self.assertEquals(val, 0)
             self.assertNotIsInstance(val, (bool, np.bool_))
 
-    def test_serializaition(self):
+    def test_update_last_sale(self):
+        metadata = {1: {'asset_type': 'equity'},
+                    2: {'asset_type': 'future',
+                        'contract_multiplier': 1000}}
+        if trading.environment is None:
```

Comment:
You can just call `TradingEnvironment.instance()` to get (or create) the current TradingEnvironment.